### DISCUSSION TITLE
Possess compile_commands.json files for intellisense analysis discussions

### DISCUSSION THREAD

**Discussion by xmdmk** [microsoft/vscode-cpptools#9962] [General]
If a project has hundreds of thousands of source code files, generated compile_commands.json, where compile_commands.json uses only a few dozen source file code, then is intellisense necessary to retrieve all the files in the working folder? I think this is the main reason for wasting system resources and causing it to crash. And I didn't find the option in the settings to do intellisense analysis only on the source code in compile_commands.json. Since the project is exported compile_commands.json files, is it necessary to analyze the source files in a working directory other than compile_commands.json? Doesn't this lead to extreme waste of system resources, greatly increasing the probability of cpptools going wrong?

**Comment by sean-mcmanus** [comment]
The compile_commands.json only has info on source files. For our global symbol search to work, we may also need to tag parse header files.

When you open a source file, IntelliSense only processes the files in the TU.

You can possibly try setting C_Cpp.files.exclude to exclude other directories you don't want to tag parse.

**Reply by xmdmk** [reply]
compile_commands.json already contains the source files required for what kind of project, in a large project, if the entire working directory is intelligently perceived, then cpptools is prone to memory leakage problems, I often encounter this phenomenon, memory is all occupied and then stuck and crashed. Currently, only clangd can be used to solve this problem. I think a setting option should be added to only analyze the files mentioned in compile_commands.json, rather than the entire working directory, which helps to reduce the resource footprint and improve operational efficiency
